src/app/tests/suites/certification/Test_TC_OO_1_1.yaml

```diff
@@ -42,63 +42,41 @@ tests:
           constraints:
               type: int16u
 
-    - label: "Step 3a: TH reads from the DUT the FeatureMap attribute."
+    - label: "Step 3: TH reads from the DUT the FeatureMap attribute."
       command: "readAttribute"
       attribute: "FeatureMap"
-      PICS: ( !OO.S.F00 && !OO.S.F01 )
       response:
-          value: 0
-          constraints:
-              type: bitmap32
+          - values:
+                - constraints:
+                      type: bitmap32
 
-    - label:
-          "Step 3b: Given OO.S.F00(LT) ensure featuremap has the correct bit set"
-      command: "readAttribute"
-      attribute: "FeatureMap"
-      PICS: OO.S.F00
-      response:
-          constraints:
-              type: bitmap32
-              hasMasksSet: [0x1]
+                - PICS: ( !OO.S.F00 && !OO.S.F01 )
```

Comment:
Why use PICS rather than having a syntax that is conditional on *actual* feature-map read or value presence?

Reply:
Well, for one things that is because it is written like this in the test plan. Test plan uses PICS and YAML usually tries to mimic the test plan.

The other thing is that the feature map may lie compared to what is declared by the vendor ?

This really depends which source of trust you use. What is declared by the vendor or what is declared by the feature map. I have no strong opinion that said and it won't hurt to read the feature map  at the beginning of each test (that needs it) and compute the PICS automatically.
But all of that is a matter of how you *get* the PICS value, not on how you use it imho.

Reply:
@raju-apple ?

Reply:
Yeah in a sense we probably will benefit by having both methods in the YAMLs , the reason I say it is because the test plans tiger team is trying out a couple of test plans in 1.3 without using PICS at all. If it works fine and the Alliance folks also agree to get rid of PICS from all test plans in general then we would have to read the feature map and deduce the PICS automatically but till then this enhancement will definitely help .

Reply:
That is fine and this is definitively orthogonal. I suspect that a simple step that reads the `FeatureMap` at the beginning of the test and convert the value into PICS code would be nice.
In this case PICS are just a shortcut name used thought the test that represent the computed results...

Reply:
I feel like I have not explained the initial reason for this PR and so we are discussing something that is not the use case about why I made it. 

Specifically this PR is made such that I can write a YAML test in a cleaner way for the `Diagnostic Logs` cluster. The diagnostic logs cluster does have conditional results in responses based on some PICS code. Those PICS code are not `FeatureMap` dependent.

Reply:
Additionaly this feature can be used to make the YAML test more readable and that is why I have updated `Test_TC_OO_1_1.

Reply:
which PICS codes is the diagnostics logs cluster using that aren't exposed in the device itself?

Reply:
The fact that the device has a wall clock or that it does support BDX or not.

Reply:
wall clock is not something that's necessarily determined a-priori and might need to be set. Given that the diagnostic logs cluster is currently provisional, the implication is that the test case has not been run previously. It seems like it would make more sense to adjust the test case to work with some of the new mechanisms we have available rather than relying more on pics. We now have a time cluster that can GIVE a device time, and a time snapshot that we can use to verify the received UTCTimeStamp together with the TimeSinceBoot. 

Re: BDX - has that been implemented in the SDK? If so, how was it tested there? Or is this the blocker for that feature landing?

Reply:
Agree with Cecille 100%. It seems like this is using a more complex way of managing coverage, rather than doing a root cause fix. If this was guaranteed to only be used for 1-2 more tests in short term, I would be more inclined to approve. However, if this pattern expands, it just will make it even harder to proceed away from PICS, which is currently desired. Please do not touch OO_1_1 and rather, bring-in the feature along with the new test that uses it. Otherwise, we cannot see the code in context of real usage.

Reply:
1.1 tests are going away anyway.

Reply:
@naveenkommareddi ?